app/widgets/common/secrets/SecretActions.ts

```diff
@@ -48,23 +59,34 @@ export default class SecretActions {
         visibility: Visibility,
         hidden: boolean,
         providerName?: string,
-        providerOptions?: ProviderOptions | null
+        providerPath?: string
     ) {
-        return this.manager.doPut(`/secrets/${key}`, {
-            body: {
-                value,
-                visibility,
-                is_hidden_value: hidden,
+        let body: SecretRequestBody = {
+            key,
+            value,
+            visibility,
+            is_hidden_value: hidden
+        };
+        if (providerName && providerPath) {
+            body = {
+                ...body,
                 provider_name: providerName,
-                provider_options: providerOptions
-            }
-        });
+                provider_options: getSecretProviderOptions(providerPath)
+            };
+        }
+        return this.manager.doPut(`/secrets/${key}`, { body });
     }
 
-    doUpdate(key: Secret['key'], value?: string, providerName?: string, providerOptions?: ProviderOptions) {
-        return this.manager.doPatch(`/secrets/${key}`, {
-            body: { value, provider_name: providerName, provider_options: providerOptions }
-        });
+    doUpdate(key: Secret['key'], value?: string, providerName?: string, providerPath?: string) {
+        let body: any = { value };
+        if (providerName && providerPath) {
+            body = {
+                ...body,
+                provider_name: providerName,
+                provider_options: getSecretProviderOptions(providerPath)
+            };
+        }
```

Comment:
Optional:

Or simply:
```suggestion
        if (providerName && providerPath) {
            body.provider_name = providerName;
            body.provider_options = getSecretProviderOptions(providerPath);
        }
```

Reply:
implemented: https://github.com/cloudify-cosmo/cloudify-stage/pull/2427/commits/0fa559952b71187d19369d99957c8610211a7d73 ✔️